### ISSUE TITLE
Custom Docker versions

### ISSUE THREAD

**Issue by simongottschlag** [ZupIT/horusec#106]
**What happened**:
When running horusec on a machine that is using Microsofts fork of moby (Azure Kubernetes Service, GitHub Actions etc) it will fail because they use other versions.

It would be great if we either could make it configurable (skip version check) or that we add a specific check for the Docker version marked with the prefix "+azure".

**What you expected to happen**:
When running horusec on a machine with Microsoft fork, it won't start:
```
2020-11-01T14:44:47.9096474Z + docker --version
2020-11-01T14:44:47.9322280Z Docker version 3.0.13+azure, build dd360c7c0de8d9132a3965db6a59d3ae74f43ba7
2020-11-01T14:44:47.9330681Z + horusec start -p=./ -e=true
2020-11-01T14:44:48.0425625Z time="2020-11-01T14:44:48Z" level=error msg="{HORUSEC_CLI} Your docker version is below of: " error=19.3
2020-11-01T14:44:48.0426489Z time="2020-11-01T14:44:48Z" level=info msg="{HORUSEC_CLI} If your docker is not installed check in docs of how to install in:\n\t\thttps://docs.docker.com/get-docker\n\t"
2020-11-01T14:44:48.0427682Z time="2020-11-01T14:44:48Z" level=panic msg="{HORUSEC_CLI} Missing required DOCKER in min. version 19.03 to start" error="check the requirements for run and try again"
2020-11-01T14:44:48.0448274Z panic: (*logrus.Entry) 0xc000362310
2020-11-01T14:44:48.0448426Z 
2020-11-01T14:44:48.0449000Z goroutine 1 [running]:
2020-11-01T14:44:48.0449589Z github.com/sirupsen/logrus.Entry.log(0xc00011c000, 0xc0005be240, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
2020-11-01T14:44:48.0450066Z 	/home/wilian/go/src/github.com/ZupIT/horusec/vendor/github.com/sirupsen/logrus/entry.go:255 +0x325
2020-11-01T14:44:48.0450391Z github.com/sirupsen/logrus.(*Entry).Log(0xc0003622a0, 0x0, 0xc0005a9da8, 0x1, 0x1)
2020-11-01T14:44:48.0450705Z 	/home/wilian/go/src/github.com/ZupIT/horusec/vendor/github.com/sirupsen/logrus/entry.go:283 +0xf0
2020-11-01T14:44:48.0451009Z github.com/sirupsen/logrus.(*Entry).Panic(0xc0003622a0, 0xc0005a9da8, 0x1, 0x1)
2020-11-01T14:44:48.0451299Z 	/home/wilian/go/src/github.com/ZupIT/horusec/vendor/github.com/sirupsen/logrus/entry.go:321 +0x55
2020-11-01T14:44:48.0452245Z github.com/ZupIT/horusec/development-kit/pkg/utils/logger.LogPanicWithLevel(0xc6283c, 0x44, 0xdd6680, 0xc000060390, 0x0, 0x0, 0x0, 0x0)
2020-11-01T14:44:48.0452789Z 	/home/wilian/go/src/github.com/ZupIT/horusec/development-kit/pkg/utils/logger/logger.go:99 +0x125
2020-11-01T14:44:48.0453311Z github.com/ZupIT/horusec/horusec-cli/internal/controllers/requirements.(*Requirements).ValidateDocker(0xc0005a9f68)
2020-11-01T14:44:48.0453872Z 	/home/wilian/go/src/github.com/ZupIT/horusec/horusec-cli/internal/controllers/requirements/requirements.go:43 +0x8b
2020-11-01T14:44:48.0454332Z main.main()
2020-11-01T14:44:48.0454969Z 	/home/wilian/go/src/github.com/ZupIT/horusec/horusec-cli/cmd/horusec/main.go:60 +0x49
2020-11-01T14:44:48.0483833Z 
2020-11-01T14:44:48.0534364Z ##[error]Bash exited with code '2'.
2020-11-01T14:44:48.0547725Z ##[section]Finishing: Code scan
```

**How to reproduce it (as minimally and precisely as possible)**:
In my case this happens on Azure Kubernetes Service, but may also be possible to reproduce on GitHub Actions.

**Anything else we need to know?**:
Read about it here: https://github.community/t/what-really-is-docker-3-0-6/16171/2

**Environment**:
- Horusec version (use `horusec version`): 1.4.0
- Operating System: Ubuntu 18.04
- Network plugin / Tool and version (if this is a network-related / tool bug): N/A
- Others: N/A

**Comment by simongottschlag** [comment]
When looking through the code, it seems like the output is taken from here: https://github.com/ZupIT/horusec/blob/c1bd1dd8911854e90ef9a1e8fc226e9ae4c89693/horusec-cli/internal/controllers/requirements/docker/docker.go#L77

If you use the following, you will get json output which may help in extracting the data: `docker version -f '{{json .}}'`.

This is the output from AKS:

```json
{
  "Client": {
    "Platform": {
      "Name": ""
    },
    "Version": "3.0.13+azure",
    "ApiVersion": "1.40",
    "DefaultAPIVersion": "1.40",
    "GitCommit": "dd360c7c0de8d9132a3965db6a59d3ae74f43ba7",
    "GoVersion": "go1.13.11",
    "Os": "linux",
    "Arch": "amd64",
    "BuildTime": "Thu May 28 20:41:25 2020",
    "Experimental": false
  },
  "Server": {
    "Platform": {
      "Name": ""
    },
    "Components": [
      {
        "Name": "Engine",
        "Version": "3.0.10+azure",
        "Details": {
          "ApiVersion": "1.40",
          "Arch": "amd64",
          "BuildTime": "Fri Jan 24 20:08:11 2020",
          "Experimental": "false",
          "GitCommit": "ea84732a77",
          "GoVersion": "go1.12.14",
          "KernelVersion": "4.15.0-1092-azure",
          "MinAPIVersion": "1.12",
          "Os": "linux"
        }
      },
      {
        "Name": "containerd",
        "Version": "v1.2.11",
        "Details": {
          "GitCommit": "f772c10a585ced6be8f86e8c58c2b998412dd963"
        }
      },
      {
        "Name": "runc",
        "Version": "1.0.0-rc10",
        "Details": {
          "GitCommit": "dc9208a3303feef5b3839f4323d9beb36df0a9dd"
        }
      },
      {
        "Name": "docker-init",
        "Version": "0.18.0",
        "Details": {
          "GitCommit": "fec3683"
        }
      }
    ],
    "Version": "3.0.10+azure",
    "ApiVersion": "1.40",
    "MinAPIVersion": "1.12",
    "GitCommit": "ea84732a77",
    "GoVersion": "go1.12.14",
    "Os": "linux",
    "Arch": "amd64",
    "KernelVersion": "4.15.0-1092-azure",
    "BuildTime": "2020-01-24T20:08:11.000000000+00:00"
  }
}
```

The below is from my local computer: (not `+azure` version)

```json
{
  "Client": {
    "Platform": {
      "Name": "Docker Engine - Community"
    },
    "Version": "19.03.12",
    "ApiVersion": "1.40",
    "DefaultAPIVersion": "1.40",
    "GitCommit": "48a66213fe",
    "GoVersion": "go1.13.10",
    "Os": "darwin",
    "Arch": "amd64",
    "BuildTime": "Mon Jun 22 15:41:33 2020",
    "Experimental": false
  },
  "Server": {
    "Platform": {
      "Name": "Docker Engine - Community"
    },
    "Components": [
      {
        "Name": "Engine",
        "Version": "19.03.12",
        "Details": {
          "ApiVersion": "1.40",
          "Arch": "amd64",
          "BuildTime": "Mon Jun 22 15:49:27 2020",
          "Experimental": "false",
          "GitCommit": "48a66213fe",
          "GoVersion": "go1.13.10",
          "KernelVersion": "4.19.76-linuxkit",
          "MinAPIVersion": "1.12",
          "Os": "linux"
        }
      },
      {
        "Name": "containerd",
        "Version": "v1.2.13",
        "Details": {
          "GitCommit": "7ad184331fa3e55e52b890ea95e65ba581ae3429"
        }
      },
      {
        "Name": "runc",
        "Version": "1.0.0-rc10",
        "Details": {
          "GitCommit": "dc9208a3303feef5b3839f4323d9beb36df0a9dd"
        }
      },
      {
        "Name": "docker-init",
        "Version": "0.18.0",
        "Details": {
          "GitCommit": "fec3683"
        }
      }
    ],
    "Version": "19.03.12",
    "ApiVersion": "1.40",
    "MinAPIVersion": "1.12",
    "GitCommit": "48a66213fe",
    "GoVersion": "go1.13.10",
    "Os": "linux",
    "Arch": "amd64",
    "KernelVersion": "4.19.76-linuxkit",
    "BuildTime": "2020-06-22T15:49:27.000000000+00:00"
  }
}
```

Not sure about the `19.03` requirement, but may be easier to match the `ApiVersion` instead of the `Version`.

**Comment by nathannascimentozup** [comment]
I agree to check the docker version on apiVersion.

The reason we limit it to version 19 is because we only tested this version and we can only guarantee that it fully works on this version.

**Comment by wiliansilvazup** [comment]
Hello, how are you @simongottschlag ?
Thanks for the informed report!
We really have a problem with the way azure works with its pipelines. For this we have some recommendations:
We put the docker in version 19.03 as a requirement, as several tests were carried out in different S.O and pipelines as suggested in the Horusec documentation, so we can guarantee that anyone who uses them in that way would have no problems with the environment of their machine.
We have some solutions that the Horusec team found your problem:
* Change the source code in order to ignore the requirements with a global flag, so no environment check will be performed on your machine, but we cannot guarantee that the analysis carried out is in accordance with our tests.
* Change your pipeline to use Horusec as a docker image. For this change, just as we do on AWS, you can start a new step in your pipeline with a Horusec-CLI image so that it will be fully configured with all the necessary requirements. The image is available on the [DockerHub](https://hub.docker.com/r/horuszup/horusec-cli)

**Comment by simongottschlag** [comment]
Hi @wiliansilvazup !

Thank you for the update!

Since we already had other problems with Docker-in-Docker and AKS, we've started moving to Virtual Machine Scale Sets instead which doesn't have these issues.

I did a test (#107) and it seems to be working as expected, but since we won't need it I'll close it. If you want to use it, feel free to take what I did and convert to checking the api version instead.

Thank you!
